drivers/serial/uart_sam0.c

```diff
@@ -89,8 +98,11 @@ static int uart_sam0_init(struct device *dev)
 	    cfg->ctrla |
 	    /* Internal clock */
 	    SERCOM_USART_CTRLA_MODE_USART_INT_CLK
+#if defined(SERCOM_USART_CTRLA_SAMPR_Msk)
```

Comment:
Use #if defined(SERCOM_USART_CTRLA_SAMPR) instead, as then you're testing for the same macro as you use.